haas_websocket/cloud_functions/api_data_CF.py

```diff
@@ -0,0 +1,77 @@
+from haas_websocket.rest.haas_rest_handler import TokenAuth
+from google.cloud import datastore
+import logging
+
+# Will run on an interval in GCP as a cloud function to regenerate the bearer token each day
+
+def parseIds (organizations):
+    if organizations:
+        ids = []
+        for item in organizations['managed_orgs']:
+            ids.append(item['id'])
+        return ids 
+    else:
+        return None
+
+def getAllThings(rest_agent, list):
+    if list:
+        things = []
+        for id in list:
+            newThings = rest_agent.getThings(id)
+            if newThings:
+                for thing in newThings['data']:
+                    things.append(thing)
+        return things
+    else:
+        return None
+
+def getAllLocations(rest_agent, list):
+    if list:
+        locations = []
+        for id in list:
+            newLocations = rest_agent.getLocations(id)
+            if newLocations:
+                for thing in newLocations['data']:
+                    locations.append(thing)
+        return locations
+    else:
+        return None
+
+def createDatastoreTasks(client, list, kind):
+    if list: 
+        tasks = []
+        appliedTasks = []
+        for entry in list:
+            if not entry["id"] in appliedTasks:
+                newTask = datastore.Entity(client.key(kind, entry["id"]))
+                newTask.update(entry)
+                appliedTasks.append(entry["id"])
+                tasks.append(newTask)
+                logging.info(entry["id"] + " added to datastore")
+            else:
+                logging.info(entry["id"] + " already in list")
+        return tasks
+    else:
+        return None
+
+# Entry point for cloud function
+def entry(request):
```

Comment:
We're doing a decent amount of logic here - fetching data, parsing ids, then storing data. Let's add some tests to handle a few cases. For instance showing what happens if we get no data back.